src/providers/token/token.ts

```diff
@@ -23,4 +25,41 @@ export class TokenProvider {
     });
   }
 
+  /**
```

Comment:
@bumi would love to get ur feedback about this function whenever ur are available.
Here we pass `array` of `token Object` and initiate `new ethers.Contract()` to each token contract and setup a `listener`.

That should work well in a `for loop` right ?